src/components/expanded-state/CustomGasState.js

```diff
@@ -0,0 +1,144 @@
+import { useIsFocused, useRoute } from '@react-navigation/native';
+import React, { useEffect, useState } from 'react';
+import { useSharedValue, withSpring } from 'react-native-reanimated';
+import { useSafeArea } from 'react-native-safe-area-context';
+import styled from 'styled-components';
+import Divider from '../Divider';
+import { ExchangeHeader } from '../exchange';
+import { FloatingPanel } from '../floating-panels';
+import { GasSpeedButton } from '../gas';
+import { Column } from '../layout';
+import { SlackSheet } from '../sheet';
+import { FeesPanel, FeesPanelTabs } from './custom-gas';
+import { getTrendKey } from '@rainbow-me/helpers/gas';
+import {
+  useAccountSettings,
+  useBooleanState,
+  useColorForAsset,
+  useDimensions,
+  useGas,
+  useHeight,
+  useKeyboardHeight,
+} from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import { colors, margin } from '@rainbow-me/styles';
+
+const springConfig = {
+  damping: 500,
+  mass: 3,
+  stiffness: 1000,
+};
+
+const FOOTER_MIN_HEIGHT = 60;
+const CONTENT_MIN_HEIGHT = 330;
+
+function useAndroidDisableGesturesOnFocus() {
+  const { params } = useRoute();
+  const isFocused = useIsFocused();
+  useEffect(() => {
+    android && params?.toggleGestureEnabled?.(!isFocused);
+  }, [isFocused, params]);
+}
+
+const FeesPanelWrapper = styled(Column)`
+  ${margin(16, 19, 29, 24)}
+`;
+
+const FeesPanelTabswrapper = styled(Column)`
+  ${margin(16, 0, 29, 0)}
+`;
+
+export default function CustomGasState({ asset }) {
+  const { network } = useAccountSettings();
+  const { setParams } = useNavigation();
+  const { params: { longFormHeight } = {} } = useRoute();
+  const { height: deviceHeight } = useDimensions();
+  const keyboardHeight = useKeyboardHeight();
+  const [isKeyboardVisible, showKeyboard, hideKeyboard] = useBooleanState();
+  const insets = useSafeArea();
+  const [footerHeight, setFooterHeight] = useHeight(FOOTER_MIN_HEIGHT);
+  const [contentHeight, setContentHeight] = useHeight(CONTENT_MIN_HEIGHT);
+  const contentScroll = useSharedValue(0);
+  const colorForAsset = useColorForAsset(asset || {}, null, false, true);
+  const { selectedGasFee, currentBlockParams } = useGas();
+  const [currentGasTrend] = useState(getTrendKey(currentBlockParams?.trend));
```

Comment:
seems interesting that this is a `useState` when there is no `setCurrentGasTrend` needed - is it better off just being a const? also, currently this means that if the trend changes while the user is on the `CustomGasState`, we won't be updating the sheet correct?

Reply:
nice yes, i saw this bug yesterday when reviewing trends 🙏